File: benchmark/perl/fib-rec.pl
sub fib {
  my $n = shift;
  if ( $n <= 2 ) { return 1; }
  return fib($n - 1) + fib($n - 2);
}


#sub fib {
#    return @_[0] < 2 ? @_[0] : fib(@_[0]-1) + fib(@_[0]-2)
#}

print fib(47)

File: benchmark/java/fibrec.java
import java.util.*;

public class fibrec {
  static long fib(long n) {
    if (n < 2) return n;
    return fib(n - 1) + fib(n - 2);
  }

  public static void main(String[] args) {
    System.out.print(fib(47));
  }
}
